### PR TITLE
Feat: Add withHover HOC

### LINKED ISSUE #472: Refactor Hoverable component to HOC
Opened by @JoseDuSV 

> Refactor Hoverable component to HOC that injects isHovered prop.

Comment by @tbergq 

> How about creating a new HOC component, then in the existing Hoverable we can add a console.warn in componentDidMount, saying this component is deprecated, please switch to new HOC component. So we can make upgrading more smooth.

(Original issue https://github.com/kiwicom/universal-components/issues/209)

### PR DESCRIPTION
**Pull request by tbergq**
Summary: withHover HOC will provide us with a better way to check
if a component is hovered. Hoverable implementation was useless
and you had to do the work in each component using it anyways.
Also added a deprecation warning to hoverable component

closes #472

### CODE REVIEW

File: packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Hoverable - native should match snapshot 1`] = `
+<View
+  onMouseEnter={[Function]}
```
Comment (JoseDuSV) [inline packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap]: When hover is available just on web, shouldn't this be just `<View>`?

File: packages/universal-components/src/WithHover/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// @flow
+
+export { default as withHover } from './withHover';
```
Comment (JoseDuSV) [inline packages/universal-components/src/WithHover/index.js]: Can you please also add export to main `src/index.js`.

Comment (tbergq) [reply packages/universal-components/src/WithHover/index.js]: Oops, I forgot that 🙈

File: packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap (hunk 1 above)
Comment (tbergq) [reply packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap]: I don't think it matters. There is no mouse on mobile. So it should never be triggered.

**Comment by tbergq** [comment]
@JoseDuSV 🚀

File: packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap (hunk 1 above)
Comment (JoseDuSV) [reply packages/universal-components/src/WithHover/__tests__/__snapshots__/withHover.test.js.snap]: True, it shouldn't break anything.

**Pull request merged**